test/code_size/hello_world_wasm2js.js

```diff
@@ -1,19 +1,30 @@
-var c = Module, g, h, k = new TextDecoder("utf8"), l;
+var d = Module, g, h, k = new TextDecoder("utf8"), l;
 
-function d(b) {
-    this.exports = function(f) {
-        function m(e) {
-            e.set = function(a, n) {
-                this[a] = n;
+function e(b) {
+    this.exports = function(r) {
+        function u(c) {
+            c.set = function(a, f) {
+                this[a] = f;
             };
-            e.get = function(a) {
+            c.get = function(a) {
                 return this[a];
             };
-            return e;
+            return c;
         }
-        return function(e) {
-            var a = new ArrayBuffer(16777216), n = e.a.a;
-            e = m([]);
+        function x(c, a, f) {
+            for (var v, p = 0, t = a, w = f.length, y = a + (3 * w >> 2) - ("=" == f[w - 2]) - ("=" == f[w - 1]); p < w; p += 4) a = m[f.charCodeAt(p + 1)], 
+            v = m[f.charCodeAt(p + 2)], c[t++] = m[f.charCodeAt(p)] << 2 | a >> 4, t < y && (c[t++] = a << 4 | v >> 2), 
+            t < y && (c[t++] = v << 6 | m[f.charCodeAt(p + 3)]);
+        }
```

Comment:
Hmm, this looks larger, doesn't it? That doesn't make sense though... but look at all this `charCodeAt` logic, it doesn't appear in the diff before?

Reply:
This is the base64 decoding logic to decode the embedded data.

Reply:
This file *is* slightly bigger, but here is no longer any `.mem` file alongside it.

If you look at the total size `hello_world_wasm2js.json` the `total` and `total_gz` both go down.

Having said that I think there will be cases where the total size goes up.. because base64 is less efficient way of storing the data.

Reply:
Oh, right, thanks - so this is indeed new code that wasn't present before. Makes sense. Tradeoff sounds fine too.